src/block-components/button/edit.js

```diff
@@ -0,0 +1,124 @@
+/*
+ * External dependencies
+ */
+import {
+	AdvancedRangeControl,
+	InspectorStyleControls,
+	PanelAdvancedSettings,
+	BaseControl,
+	AdvancedToolbarControl,
+	ColorPaletteControl,
+	FourRangeControl,
+} from '~stackable/components'
+import { i18n } from 'stackable'
+import {
+	useBlockEl, useBlockHoverState, useAttributeEditHandlers,
+} from '~stackable/hooks'
+
+/**
+ * Internal dependencies
+ */
+import { BorderControls } from '../helpers'
+import { LinkControls } from '../helpers/link'
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n'
+import { Fragment } from '@wordpress/element'
+
+export const Edit = () => {
+	const [ hoverState ] = useBlockHoverState()
+	const blockEl = useBlockEl()
+
+	const {
+		getAttribute,
+	} = useAttributeEditHandlers()
+
+	return (
+		<InspectorStyleControls>
+			<PanelAdvancedSettings
+				title={ __( 'Link', i18n ) }
+				id="link"
+			>
+				<LinkControls attrNameTemplate="link%s" />
+			</PanelAdvancedSettings>
+
+			<PanelAdvancedSettings
+				title={ __( 'Button', i18n ) }
+				id="button"
+			>
+
+				<Fragment>
+					<BaseControl
+						id="stk-button-normal-color-type"
+					>
+						<Fragment>
+							<AdvancedToolbarControl
+								controls={ [
+									{
+										value: '',
+										title: __( 'Single', i18n ),
+									},
+									{
+										value: 'gradient',
+										title: __( 'Gradient', i18n ),
+									},
+								] }
+								attribute="buttonBackgroundColorType"
+								isSmall={ true }
+								fullwidth={ false }
+								hover="all"
+							/>
+							<ColorPaletteControl
+								label={ getAttribute( 'buttonBackgroundColorType', 'desktop', hoverState ) === 'gradient'
+									? sprintf( __( 'Button Color #%s', i18n ), 1 )
+									: __( 'Button Color', i18n )
+								}
+								attribute="buttonBackgroundColor"
+								hasTransparent={ getAttribute( 'buttonBackgroundColorType', 'desktop', hoverState ) === 'gradient' !== 'gradient' }
+								hover="all"
+							/>
+							{ getAttribute( 'buttonBackgroundColorType', 'desktop', hoverState ) === 'gradient' && (
+								<Fragment>
+									<ColorPaletteControl
+										label={ __( 'Button Color #2', i18n ) }
+										attribute="buttonBackgroundColor2"
+										hover="all"
+									/>
+
+									<AdvancedRangeControl
+										label={ __( 'Gradient Direction (degrees)', i18n ) }
+										attribute="buttonBackgroundGradientDirection"
+										min={ 0 }
+										max={ 360 }
+										step={ 10 }
+										allowReset={ true }
+										hover="all"
+									/>
+								</Fragment>
+							) }
+						</Fragment>
+					</BaseControl>
+
+					<BorderControls
+						blockEl={ blockEl }
```

Comment:
Please note that I'm still not sure whether we should keep using `blockEl`. Keep this for now, but we'll decide on a better way to do this in the future